### PR TITLE
feature(#620) : create database

### PR DESCRIPTION
**Pull request by Subsegment**
# Which issue does this PR close?

Closes #.

# Rationale for this change

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: common/models/src/schema.rs [hunk 1]
```diff
@@ -267,3 +268,127 @@ impl ColumnType {
...
     }
 }
+
+#[derive(Debug, Clone)]
+pub struct DatabaseSchema {
+    pub name: String,
+    pub config: DatabaseOptions,
+}
+
+impl SchemaProvider for DatabaseSchema {
```
Comment (yukkit) [inline common/models/src/schema.rs:278]: move to UserSchema

File: common/models/src/schema.rs [hunk 2]
```diff
@@ -267,3 +268,127 @@ impl ColumnType {
...
+pub enum Precision {
+    MS,
+    US,
+    NS,
+}
+
+impl Precision {
+    pub fn new(text: &str) -> Option<Self> {
+        match text.to_uppercase().as_str() {
+            "'MS'" => Some(Precision::MS),
```
Comment (yukkit) [inline common/models/src/schema.rs:347]: remove single quote

File: common/models/src/schema.rs [hunk 3]
```diff
@@ -267,3 +268,127 @@ impl ColumnType {
...
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Duration {
+    pub time_num: u64,
+    pub unit: DurationUnit,
+}
+
+impl Duration {
+    // text should have ', such as '10d', the len of '10d' is 5
+    pub fn new(text: &str) -> Option<Self> {
+        let len = text.len();
```
Comment (yukkit) [inline common/models/src/schema.rs:371]: relation single quote

File: query_server/query/src/sql/parser.rs [hunk 4]
```diff
@@ -19,6 +22,12 @@ use trace::debug;
 const TAGS: &str = "TAGS";
 const CODEC: &str = "CODEC";
 
+const TTL: &str = "TTL";
```
Comment (yukkit) [inline query_server/query/src/sql/parser.rs:25]: Token

File: query_server/spi/src/query/ast.rs [hunk 5]
```diff
@@ -58,6 +62,32 @@ pub struct ColumnOption {
     pub codec: String,
 }
 
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct DatabaseOptions {
```
Comment (yukkit) [inline query_server/spi/src/query/ast.rs:66]: Optional

File: query_server/query/src/sql/planner.rs [hunk 6]
```diff
@@ -339,6 +343,65 @@ impl<S: ContextProvider> SqlPlaner<S> {
...
+            plan_options.shard_num = options.shard_num
+        }
+        if options.vnode_duration != String::default() {
+            plan_options.vnode_duration = self.str_to_duration(&options.vnode_duration)?;
+        }
+        if options.precision != String::default() {
+            plan_options.precision = match Precision::new(&options.precision) {
+                None => {
+                    return Err(LogicalPlannerError::External {
+                        source: DataFusionError::Internal(format!(
```
Comment (yukkit) [inline query_server/query/src/sql/planner.rs:378]: use ourself err

File: query_server/query/src/sql/planner.rs [hunk 7]
```diff
@@ -339,6 +343,65 @@ impl<S: ContextProvider> SqlPlaner<S> {
...
+            }
+        }
+        Ok(plan_options)
+    }
+
+    fn str_to_duration(&self, text: &str) -> Result<Duration> {
+        let duration = match Duration::new(text) {
+            None => {
+                return Err(LogicalPlannerError::External {
+                    source: DataFusionError::Internal(format!(
```
Comment (yukkit) [inline query_server/query/src/sql/planner.rs:394]: DataFusionError::Internal

**Comment by Subsegment** [comment]
we need optimize token and fix wrong create database when create table

**Review by yukkit — APPROVED** [review]

**Pull request merged**